### PR TITLE
Update NFT state for new requirements

### PR DESCRIPTION
**Pull request by azoyan**

### CODE REVIEW

File: src/contract.rs [hunk 1]
```diff
@@ -223,13 +225,14 @@ pub fn get_hash(account: &ActorId, transaction_id: u64) -> H256 {
...
-    fn from(value: &Nft) -> Self {
-        let Nft {
+impl From<&Contract> for IoNFT {
+    fn from(value: &Contract) -> Self {
+        let Contract {
             token,
             token_id,
             owner,
             transactions,
+            collection: _,
```
Comment (fluiderson) [inline src/contract.rs:235]:
```suggestion
            ...
```

**Review by fluiderson — APPROVED** [review]

File: src/contract.rs [hunk 2]
```diff
@@ -3,34 +3,40 @@ use gear_lib_derive::{NFTCore, NFTMetaState, NFTStateKeeper};
...
     }
-    let nft = Nft {
+    let nft = Contract {
         token: NFTState {
-            name: config.name,
-            symbol: config.symbol,
-            base_uri: config.base_uri,
+            name: config.collection.name.clone(),
+            symbol: "".to_string(),
+            base_uri: "".to_string(),
```
Comment (shamilsan) [inline src/contract.rs:34]: An empty string is a default I guess.
```suggestion
```

File: src/contract.rs [hunk 3]
```diff
@@ -191,6 +212,33 @@ impl Nft {
         );
         self.transactions.remove(&transaction_hash);
     }
+
+    fn check_constraints(&self) {
+        // gstd::debug!("self.constraints: {:?}", self.constraints);
```
Comment (shamilsan) [inline src/contract.rs:217]: Dead code
```suggestion
```

File: src/contract.rs [hunk 4]
```diff
@@ -191,6 +212,33 @@ impl Nft {
...
+            if max_mint_count <= self.token.token_metadata_by_id.len() as u32 {
+                panic!(
+                    "Mint impossible because max minting count {} limit exceeded",
+                    max_mint_count
+                );
+            }
+        }
+
+        let current_minter = msg::source();
+        // gstd::debug!("current_minter: {:?}", current_minter);
```
Comment (shamilsan) [inline src/contract.rs:228]: Also,
```suggestion
```

File: src/contract.rs [hunk 5]
```diff
@@ -244,3 +293,53 @@ impl From<&Nft> for IoNFT {
...
+            .map(|(hash, event)| (*hash, event.clone()))
+            .collect();
+
+        let token_metadata_by_id = token
+            .token_metadata_by_id
+            .iter()
+            .map(|(id, metadata)| {
+                let metadata = metadata.as_ref().unwrap();
+                let nft = Nft {
+                    owner: *token.owner_by_id.get(id).unwrap(),
```
Comment (shamilsan) [inline src/contract.rs:325]: `get` + `unwrap` often can be replaced by `[]`
```suggestion
                    owner: *token.owner_by_id[id],
```

**Review by shamilsan — APPROVED** [review]

**Pull request merged**